File: packages/shared/src/types/membership.ts
export type MembershipStatus = "NONE" | "PENDING" | "ACTIVE" | "EXPIRED" | "CANCELED";

export type MembershipPlan = "MONTHLY" | "YEARLY";

export interface MembershipPlanInfo {
  plan: MembershipPlan;
  label: string;
  priceLabel: string;
}

/** Gösterim amaçlı — gerçek fiyat/plan Iyzico panelinde tanımlanır. */
export const MEMBERSHIP_PLANS: MembershipPlanInfo[] = [
  { plan: "MONTHLY", label: "Aylık", priceLabel: "499 ₺ / ay" },
  { plan: "YEARLY", label: "Yıllık", priceLabel: "4.990 ₺ / yıl (2 ay bedava)" },
];

export interface CompanyMembershipDto {
  status: MembershipStatus;
  plan?: MembershipPlan | null;
  expiresAt?: string | null;
  phoneVerified: boolean;
  /** Gerçek ödeme durumundan bağımsız olarak şu an tam erişimi var mı (beta dönemi dahil). */
  isPremium: boolean;
  /** Erişimin ödenmiş bir üyelikten değil, Erken Erişim/Beta döneminden geldiğini belirtir. */
  betaFreeAccess: boolean;
}

export interface InitiateCheckoutInput {
  plan: MembershipPlan;
}

/** Bir kullanıcının geçmiş üyelik/ödeme kaydı — Hesabım panelindeki "Faturalarım" listesinde gösterilir. */
export interface MembershipHistoryEntry {
  id: string;
  plan: MembershipPlan;
  /** Ödeme anındaki plan adı/fiyat metninin anlık görüntüsü — ileride fiyat değişse bile geçmiş kayıt değişmez. */
  planLabel: string | null;
  planPriceLabel: string | null;
  status: MembershipStatus;
  billingContactName: string | null;
  billingCity: string | null;
  billingAddress: string | null;
  billingZipCode: string | null;
  startedAt: string | null;
  currentPeriodEnd: string | null;
  canceledAt: string | null;
  createdAt: string;
}

export interface InitiateCheckoutResponse {
  checkoutFormContent: string;
  token: string;
}

export interface SendPhoneCodeInput {
  phone: string;
}

export interface VerifyPhoneCodeInput {
  code: string;
}

File: apps/mobile/lib/models/membership.dart
class CompanyMembership {
  final String status;
  final String? plan;
  final String? expiresAt;
  final bool phoneVerified;
  final bool isPremium;
  final bool betaFreeAccess;

  const CompanyMembership({
    required this.status,
    this.plan,
    this.expiresAt,
    required this.phoneVerified,
    required this.isPremium,
    required this.betaFreeAccess,
  });

  factory CompanyMembership.fromJson(Map<String, dynamic> json) => CompanyMembership(
        status: json['status'] as String,
        plan: json['plan'] as String?,
        expiresAt: json['expiresAt'] as String?,
        phoneVerified: json['phoneVerified'] as bool,
        isPremium: json['isPremium'] as bool? ?? false,
        betaFreeAccess: json['betaFreeAccess'] as bool? ?? false,
      );
}

class MembershipPlanInfo {
  final String plan;
  final String label;
  final String priceLabel;

  const MembershipPlanInfo({required this.plan, required this.label, required this.priceLabel});

  factory MembershipPlanInfo.fromJson(Map<String, dynamic> json) => MembershipPlanInfo(
        plan: json['plan'] as String,
        label: json['label'] as String,
        priceLabel: json['priceLabel'] as String,
      );
}

class CheckoutResult {
  final String checkoutFormContent;
  final String token;

  const CheckoutResult({required this.checkoutFormContent, required this.token});

  factory CheckoutResult.fromJson(Map<String, dynamic> json) => CheckoutResult(
        checkoutFormContent: json['checkoutFormContent'] as String,
        token: json['token'] as String,
      );
}